src/Compilers/CSharp/Test/Syntax/LexicalAndXml/SyntaxTokenParserTests.cs

```diff
@@ -265,6 +266,119 @@ public void ResultContextualKind()
         AssertToken(expectedKind: SyntaxKind.ClassKeyword, expectedContextualKind: SyntaxKind.None, new TextSpan(16, 5), "class", parser.ParseNextToken());
     }
 
+    [Fact]
+    public void ParseLeadingTrivia_Empty()
+    {
+        var sourceText = SourceText.From("class C { }");
+        var parser = SyntaxFactory.CreateTokenParser(sourceText, TestOptions.Regular);
+
+        var result = parser.ParseLeadingTrivia();
+        AssertToken(expectedKind: SyntaxKind.None, expectedContextualKind: SyntaxKind.None, new TextSpan(0, 0), "", result);
+        Assert.Empty(result.Token.LeadingTrivia);
+        Assert.Empty(result.Token.TrailingTrivia);
+
+        AssertToken(expectedKind: SyntaxKind.ClassKeyword, expectedContextualKind: SyntaxKind.None, new TextSpan(0, 6), "class ", parser.ParseNextToken());
+    }
+
+    [Fact]
+    public void ParseLeadingTrivia_SameLine()
+    {
+        var sourceText = SourceText.From("/* test */ class C { }");
+        var parser = SyntaxFactory.CreateTokenParser(sourceText, TestOptions.Regular);
+
+        var result = parser.ParseLeadingTrivia();
+        AssertToken(expectedKind: SyntaxKind.None, expectedContextualKind: SyntaxKind.None, new TextSpan(0, 11), "/* test */ ", result);
+        AssertTrivia(result.Token.LeadingTrivia,
+            (SyntaxKind.MultiLineCommentTrivia, "/* test */"),
+            (SyntaxKind.WhitespaceTrivia, " "));
+        Assert.Empty(result.Token.TrailingTrivia);
+
+        AssertToken(expectedKind: SyntaxKind.ClassKeyword, expectedContextualKind: SyntaxKind.None, new TextSpan(11, 6), "class ", parser.ParseNextToken());
+
+        parser.ResetTo(result);
+        AssertToken(expectedKind: SyntaxKind.ClassKeyword, expectedContextualKind: SyntaxKind.None, new TextSpan(0, 17), "/* test */ class ", parser.ParseNextToken());
```

Comment:
Consider testing calls to `parser.ParseLeadingTrivia()` and to `parser.ParseNextToken()` without first calling `parser.ResetTo(result)`.